packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1539,14 +1446,19 @@ export class KeyringController extends BaseController<
     removedAccounts: string[];
     remainingAccounts: string[];
   }> {
-    const keyring = await this.getOrAddQRKeyring();
+    const keyring = this.getQRKeyring();
+
+    if (!keyring) {
+      return { removedAccounts: [], remainingAccounts: [] };
+    }
+
```

Comment:
By avoiding an eventual call to `#addQRKeyring`, we can wrap everything in the mutex lock in the subsequent PR: https://github.com/MetaMask/core/pull/4182/files#diff-cacd62f731d7f6947fe841f6ba3441c7a16f764b699801b0c212e0e6c15879bfR1471

Also, it doesn't make sense to add a QR Keyring for the sole purpose of removing it

Reply:
We can alternatively keep the exact same behavior with the same trick used in other QR functions to avoid deadlock:
```typescript
const keyring = this.getQRKeyring() || (await this.#addQRKeyring());
```